source/extensions/filters/http/basic_auth/basic_auth_filter.cc

```diff
@@ -111,7 +111,10 @@ Http::FilterHeadersStatus BasicAuthFilter::onDenied(absl::string_view body,
       Http::Code::Unauthorized, body,
       [this](Http::ResponseHeaderMap& headers) {
         const auto request_headers = this->decoder_callbacks_->requestHeaders();
-        const std::string uri = Http::Utility::buildOriginalUri(*request_headers, MaximumUriLength);
+        const std::string uri =
+            request_headers.has_value()
+                ? Http::Utility::buildOriginalUri(*request_headers, MaximumUriLength)
```

Comment:
@zhaohuabing What should the filter do in case of an empty request headers?

Reply:
I don't known in which scenarios the request headers would be nil. If it does, would it be a bug instead of an error?

Reply:
@adisuissa @antoniovleonti I'm fine with using an empty string here, but it would be useful to add a comment explainning this scenarion occurs only during testing.

Should we add an assert here to ensure this won't result in an nullpointer and fix the tests instead?